### PR TITLE
Add blosc as compressor

### PR DESCRIPTION
**Pull request by mrocklin**
This sometimes performs better or worse than LZ4.  

Currently I'm leaning towards including it but keeping LZ4 as default.

### CODE REVIEW

File: distributed/protocol.py [hunk 1]
```diff
@@ -63,6 +63,14 @@
     default_compression = 'lz4'
 
 
+with ignoring(ImportError):
+    import blosc
+    blosc.set_nthreads(2)
```
Comment (ogrisel) [inline distributed/protocol.py]: Have you tried to play with increasing this to see its impact on compression speed?

**Comment by ogrisel** [comment]
I think LZ4 is a reasonable default to but AFAIK it does not support multi-threading while blosc does. Not sure if multithreading helps so much in practice.

File: distributed/protocol.py (hunk 1 above)
Comment (mrocklin) [inline distributed/protocol.py]: I found that benefits of compression fall off a bit after the first few cores.  I'm also concerned about cores fighting for resources.  Often our cores are busy doing actual work.

**Comment by mrocklin** [comment]
My plan is to leave this PR as is until I run into a problem where compression significantly helps and then do some benchmarking against that and other problems.

**Comment by mrocklin** [comment]
Closing this for now.  I haven't yet run into a case where the performance difference warrants the added complexity.

**Pull request closed without merging**